Jenkinsfile

```diff
@@ -0,0 +1,259 @@
+                            // findbugs publishing is skipped in both steps because multi-module projects cause
+                            // extra copies to be reported. Instead, use commonPost to collect once at the end
+                            // of the build.
+                            writeFile file:'extract-iam-credential.sh', text:libraryResource('scripts/extract-iam-credential.sh')
+                            sh '''
+                                bash extract-iam-credential.sh
+
+                                # Hide login credential from below
+                                set +x
+
+                                LOGIN_CMD=$(aws ecr get-login --no-include-email --region us-west-2)
+
+                                $LOGIN_CMD
+                            '''
+                            sh '''
+                                echo $ARTIFACTORY_PASSWORD | docker login confluent-docker.jfrog.io -u $ARTIFACTORY_USERNAME --password-stdin
+                            '''
+                            writeFile file:'create-pip-conf-with-nexus.sh', text:libraryResource('scripts/create-pip-conf-with-nexus.sh')
+                            writeFile file:'create-pypirc-with-nexus.sh', text:libraryResource('scripts/create-pypirc-with-nexus.sh')
+                            writeFile file:'setup-credential-store.sh', text:libraryResource('scripts/setup-credential-store.sh')
+                            writeFile file:'set-global-user.sh', text:libraryResource('scripts/set-global-user.sh')
+                            sh '''
+                                bash create-pip-conf-with-nexus.sh
+                                bash create-pypirc-with-nexus.sh
+                                bash setup-credential-store.sh
+                                bash set-global-user.sh
+                            '''
+
+                            // Set the project versions in the pom files
+                            sh "mvn --batch-mode versions:set -DnewVersion=${config.ksql_db_version} -DgenerateBackupPoms=false"
+
+                            // Set the version of the parent project to use.
+                            sh "mvn --batch-mode versions:update-parent -DparentVersion=\"[${config.cp_version}]\" -DgenerateBackupPoms=false"
+
+                            if (config.release) {
+                                def git_tag = "v${config.ksql_db_version}-ksqldb"
+                                sh "git add ."
+                                sh "git commit -m \"Setting project version and parent version.\""
+                                sh "git tag ${git_tag}"
+                                sh "git push -f origin ${git_tag}"
+                            }
+
+                            cmd = "mvn --batch-mode -Pjenkins clean package dependency:analyze site validate -U "
+                            cmd += "-DskipTests "
```

Comment:
let's not forget to get rid of these things!

Reply:
yup